lib/publiccloud/ec2.pm

```diff
@@ -49,11 +51,20 @@ sub init {
         $self->vault_create_credentials();
     }
 
+    die('Credentials are invalid') unless ($self->_check_credentials());
+
     assert_script_run("export AWS_ACCESS_KEY_ID=" . $self->key_id);
     assert_script_run("export AWS_SECRET_ACCESS_KEY=" . $self->key_secret);
     assert_script_run('export AWS_DEFAULT_REGION="' . $self->region . '"');
 
-    die('Credentials are invalid') unless ($self->_check_credentials());
+    if (get_var('PUBLIC_CLOUD_SLES4SAP')) {
+        my $credentials_file = "[default]" . $/
+          . 'aws_access_key_id=' . $self->key_id . $/
+          . 'aws_secret_access_key=' . $self->key_secret;
+
+        save_tmp_file(CREDENTIALS_FILE, $credentials_file);
+        assert_script_run('curl -O ' . autoinst_url . "/files/" . CREDENTIALS_FILE);
```

Comment:
According to  https://github.com/SUSE/ha-sap-terraform-deployments/tree/master/aws you need to place this file to  `$HOME/.aws/credentials`, or?

Reply:
it's because of this line in my config file: `aws_credentials = "/root/ec2_credentials"`, but indeed could be changed. I just used the same behaviour as for GCP.

Reply:
ah, perfect then. I just wasn't aware that you also configure it.

Reply:
And in fact I should change the variable in my config file rather than fixing it, otherwise I will have to change the value twice if it is changed in this code, so I will do a small fix here.